plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -286,7 +288,23 @@ public PreparedStatement getPreparedStatement(Connection connection, String sql)
     @Override
     public TableStatistics getTableStatistics(ConnectorSession session, JdbcTableHandle handle, TupleDomain<ColumnHandle> tupleDomain)
     {
-        return delegate.getTableStatistics(session, handle, tupleDomain);
+        if (!handle.isNamedRelation()) {
+            return delegate.getTableStatistics(session, handle, tupleDomain);
+        }
+
+        // session stays the same, assuming no row filtering
```

Comment:
Please improve the comment here (what does it mean that `session stays the same`?)